#!/usr/bin/env python
PACKAGE = "rov_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kp_x", double_t, 1, "x kp",    1, 0,   200)
gen.add("ki_x", double_t, 1, "x ki",    0, 0,   200)
gen.add("kd_x", double_t, 1, "x kd",    0, 0,   200)
gen.add("kp_y", double_t, 1, "y kp",    1, 0,   200)
gen.add("ki_y", double_t, 1, "y ki",    0, 0,   200)
gen.add("kd_y", double_t, 1, "y kd",    0, 0,   200)
gen.add("kp_w", double_t, 1, "w kp",    1, 0,   200)
gen.add("ki_w", double_t, 1, "w ki",    0, 0,   200)
gen.add("kd_w", double_t, 1, "w kd",    0, 0,   200)
gen.add("kp_depth", double_t, 1, "depth kp",    1, 0,   200)
gen.add("ki_depth", double_t, 1, "depth ki",    0, 0,   200)
gen.add("kd_depth", double_t, 1, "depth kd",    0, 0,   200)
gen.add("kp_roll", double_t, 1, "roll kp",    1, 0,   200)
gen.add("ki_roll", double_t, 1, "roll ki",    0, 0,   200)
gen.add("kd_roll", double_t, 1, "roll kd",    0, 0,   200)
gen.add("floatability", double_t, 1, "floatability",    1, 0,   200)

exit(gen.generate(PACKAGE, "rov_control", "pid"))